File: data-ingestion/src/db/models/track.py
"""Track model."""
from sqlalchemy import Column, String, Integer
from sqlalchemy.orm import relationship
from src.db.base import Base


class Track(Base):
    """Racing track."""

    __tablename__ = "tracks"
    __table_args__ = {'schema': 'racing'}

    id = Column(Integer, primary_key=True, autoincrement=True)
    track_id = Column(String(10), unique=True, nullable=False, index=True)
    track_name = Column(String(100), nullable=False)
    country = Column(String(10))

    # Relationships
    meets = relationship("Meet", back_populates="track", cascade="all, delete-orphan")

    def __repr__(self):
        return f"<Track(track_id='{self.track_id}', name='{self.track_name}')>"

File: backend-api/src/main/java/com/horseracing/api/entity/Track.java
package com.horseracing.api.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "tracks", schema = "racing")
@Data
@NoArgsConstructor
public class Track {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "track_id", nullable = false, unique = true)
    private String trackId;

    @Column(name = "track_name", nullable = false)
    private String trackName;

    @Column(name = "country")
    private String country;
}